### PR TITLE
Expose serial number & device type properties

### PR DESCRIPTION
**Pull request by imgrant**
Routines taken from [Chris Balmer's Si7021 MicroPython driver](https://github.com/chrisbalmer/micropython-si7021).

### CODE REVIEW

File: adafruit_si7021.py [hunk 1]
```diff
@@ -141,3 +143,66 @@ def start_measurement(self, what):
         elif self._measurement != what:
             raise RuntimeError("other measurement in progress")
         self._measurement = what
+
+    @property
+    def serial_number(self):
+        """The device's unique ID (serial number)."""
+        serial, _ = self._get_device_info()
```
Comment (ladyada) [inline adafruit_si7021.py]: return self._get_device_info()[0]

File: adafruit_si7021.py [hunk 2]
```diff
@@ -141,3 +143,66 @@ def start_measurement(self, what):
...
+        return serial
+    
+    @property
+    def device_identifier(self):
+        """A device identifier (model type) string."""
+        _, identifier = self._get_device_info()
+        return identifier
```
Comment (ladyada) [inline adafruit_si7021.py]: return self._get_device_info()[1]

**Review by ladyada — CHANGES_REQUESTED** [review]

**Comment by ladyada** [comment]
please check the suggestions, also run pylint and black :)
https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

File: adafruit_si7021.py (hunk 1 above)
Comment (imgrant) [reply adafruit_si7021.py]: Thanks, fixed in 6587e8749b9aba6e66101219cd3dd706613214f7

**Comment by imgrant** [comment]
> please check the suggestions, also run pylint and black :)
> https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

Thanks, I've pushed the return & formatting changes
Edit: I spoke too soon, I will fixup the pylint complaints that I now see in the CI output

**Comment by ladyada** [comment]
please check CI and the guide until you get a green check - then tag me for a final review :)

**Comment by imgrant** [comment]
Thanks for the tips @ladyada CI has passed now!

**Comment by ladyada** [comment]
@imgrant awesome, can you tell me what hardware you used to test?

**Comment by imgrant** [comment]
> @imgrant awesome, can you tell me what hardware you used to test?

An Adafruit Si7021 breakout board (the older generation, without STEMMA QT/Qwic connectors), and a Raspberry Pi (3B)

**Comment by ladyada** [comment]
kk im not going to test with hardware cause it looks OK to me, we'll see if anyone has issues!

**Comment by anecdata** [comment]
Tested on PyPortal with Stemma QT version of Si7021, looks good.

**Comment by ladyada** [comment]
thank you @anecdata!

**Pull request merged**